upstream/ami.go

```diff
@@ -50,13 +55,14 @@ type AMI struct {
 // `~/.aws/config` and `~/.aws/credentials` files, and support environment variables.
 // See AWS documentation for more details:
 // https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/sessions.html
-func NewAWSClient() *ec2.EC2 {
+func NewAWSClient() *ec2.Client {
 	// Create a new session based on shared / env credentials
-	s := session.Must(session.NewSessionWithOptions(session.Options{
-		SharedConfigState: session.SharedConfigEnable,
-	}))
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		log.Fatal("failed to load aws config", err)
+	}
```

Comment:
I believe the v2 behaviour of `config.LoadDefaultConfig` is identical to our previous setting of explicitly setting `session.SharedConfigEnable`, as in it'll look into ~/.aws/config and ~/.aws/credentials. At least at I can confirm it works on my machine, correctly picking up config from ~/.aws/config (grabbing creds via a `credential_process` so not the most straightforward config!).